### PR TITLE
Update jni to remove row conversion code

### PR DESCRIPTION
**Pull request by ttnghia**
This removes the row conversion code as it has been moved into libcudf (https://github.com/rapidsai/cudf/pull/14664) to overcome a CCCL issue (https://github.com/NVIDIA/spark-rapids-jni/issues/1567).

We may need to move the code back from libcudf once the root cause of the issue is fixed.

### CODE REVIEW

**Comment by ttnghia** [comment]
Depends on https://github.com/rapidsai/cudf/pull/14664.

File: src/main/cpp/src/RowConversionJni.cpp [hunk 1]
```diff
@@ -21,97 +21,97 @@
...
   JNI_NULL_CHECK(env, input_table, "input table is null", 0);
 
   try {
     cudf::jni::auto_set_device(env);
-    cudf::table_view const* n_input_table = reinterpret_cast<cudf::table_view const*>(input_table);
+    cudf::table_view const *n_input_table =
+        reinterpret_cast<cudf::table_view const *>(input_table);
     std::vector<std::unique_ptr<cudf::column>> cols =
-      spark_rapids_jni::convert_to_rows_fixed_width_optimized(*n_input_table);
+        cudf::jni::convert_to_rows_fixed_width_optimized(*n_input_table);
```
Comment (hyperbolic2346) [inline src/main/cpp/src/RowConversionJni.cpp:33]:
```suggestion
        cudf::convert_to_rows_fixed_width_optimized(*n_input_table);
```

File: src/main/cpp/src/RowConversionJni.cpp [hunk 2]
```diff
@@ -21,97 +21,97 @@
...
   JNI_NULL_CHECK(env, input_table, "input table is null", 0);
 
   try {
     cudf::jni::auto_set_device(env);
-    cudf::table_view const* n_input_table = reinterpret_cast<cudf::table_view const*>(input_table);
+    cudf::table_view const *n_input_table =
+        reinterpret_cast<cudf::table_view const *>(input_table);
     std::vector<std::unique_ptr<cudf::column>> cols =
-      spark_rapids_jni::convert_to_rows(*n_input_table);
+        cudf::jni::convert_to_rows(*n_input_table);
```
Comment (hyperbolic2346) [inline src/main/cpp/src/RowConversionJni.cpp:53]:
```suggestion
        cudf::convert_to_rows(*n_input_table);
```

File: src/main/cpp/src/RowConversionJni.cpp [hunk 3]
```diff
@@ -21,97 +21,97 @@
...
-                   n_scale.begin(),
-                   std::back_inserter(types_vec),
-                   [](jint type, jint scale) { return cudf::jni::make_data_type(type, scale); });
+    std::transform(n_types.begin(), n_types.end(), n_scale.begin(),
+                   std::back_inserter(types_vec), [](jint type, jint scale) {
+                     return cudf::jni::make_data_type(type, scale);
+                   });
     std::unique_ptr<cudf::table> result =
-      spark_rapids_jni::convert_from_rows(list_input, types_vec);
+        cudf::jni::convert_from_rows(list_input, types_vec);
```
Comment (hyperbolic2346) [inline src/main/cpp/src/RowConversionJni.cpp:114]:
```suggestion
        cudf::convert_from_rows(list_input, types_vec);
```

File: src/main/cpp/src/RowConversionJni.cpp [hunk 4]
```diff
@@ -21,97 +21,97 @@
...
-                   n_scale.begin(),
-                   std::back_inserter(types_vec),
-                   [](jint type, jint scale) { return cudf::jni::make_data_type(type, scale); });
+    std::transform(n_types.begin(), n_types.end(), n_scale.begin(),
+                   std::back_inserter(types_vec), [](jint type, jint scale) {
+                     return cudf::jni::make_data_type(type, scale);
+                   });
     std::unique_ptr<cudf::table> result =
-      spark_rapids_jni::convert_from_rows_fixed_width_optimized(list_input, types_vec);
+        cudf::jni::convert_from_rows_fixed_width_optimized(list_input,
```
Comment (hyperbolic2346) [inline src/main/cpp/src/RowConversionJni.cpp:85]:
```suggestion
        cudf::convert_from_rows_fixed_width_optimized(list_input,
```

**Review by hyperbolic2346 — CHANGES_REQUESTED** [review]

**Comment by ttnghia** [comment]
build

**Comment by hyperbolic2346** [comment]
Will this work before the proclaim_return_type changes go in?

**Comment by ttnghia** [comment]
Oh yeah, this is a good question. The answer is NO. It seems that this PR and https://github.com/NVIDIA/spark-rapids-jni/pull/1662 needs to be combined.

**Comment by ttnghia** [comment]
build

**Comment by ttnghia** [comment]
Close as this is combined into https://github.com/NVIDIA/spark-rapids-jni/pull/1662.

**Pull request closed without merging**